controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/pricing/PricingApiHandler.java

```diff
@@ -138,8 +179,44 @@ private ClusterResources clusterResources(String resourcesString) {
         return new ClusterResources(nodes, 1, nodeResources);
     }
 
-    private List<Pair<String, String>> keysAndValues(String[] elements) {
-        return Arrays.stream(elements).map(element -> {
+    private ApplicationResources applicationResources(String appResourcesString) {
+        List<String> elements = Arrays.stream(appResourcesString.split(",")).toList();
+
+        var applicationName = "default";
+        var vcpu = 0d;
+        var memoryGb = 0d;
+        var diskGb = 0d;
+        var gpuMemoryGb = 0d;
+        var enclaveVcpu = 0d;
+        var enclaveMemoryGb = 0d;
+        var enclaveDiskGb = 0d;
+        var enclaveGpuMemoryGb = 0d;
+
+        for (var element : keysAndValues(elements)) {
+            switch (element.getFirst().toLowerCase()) {
+                case "name" -> applicationName = element.getSecond();
+
+                case "vcpu" -> vcpu = parseDouble(element.getSecond());
```

Comment:
Do we need to go via `double` here` or can this be parsed straight to `BigDecimal`?

Reply:
Yeah, I thought about that as well earlier today, will look into it in a later PR